doc/chips/9.rst

```diff
@@ -45,36 +49,285 @@ The idea of ``mason`` is based heavily off of Rust's cargo_. It is meant to
+of dependencies. Finally, if the package is an application, the binary will
+build into ``target/``. The final directory hierarchy will look like this:
+
+::
+
+  ~/.mason/src/
+    (dependencies of MyPackage)
+
+
+  MyPackage/
+    Mason.toml
+    Mason.lock
+    src/
+      main.chpl
+    target/
+      ( MyPackage )
+
+
+Project Code
+++++++++++++
+
+This is the Chapel source code written by the package author.
+
+For example, ``main.chpl``:
+
+.. code-block:: chapel
+
+   proc main() {
+       writeln("Hello, world!");
+    }
+
+Project Code -> Manifest File
++++++++++++++++++++++++++++++
+
+The user will be responsible for writing the manifest file in its entirety.
+
+Manifest File
++++++++++++++
+
+The ``Mason.toml`` manifest file will closely follow the design of
+`Cargo's manifest file <http://doc.crates.io/manifest.html>`_,
+``Cargo.toml``. Some fields will vary, if they are too Rust-specific or could
+benefit from being more Chapel-specific.
+
+For example, ``Mason.toml``:
+
+.. code-block:: ini
+
+   [package]
+   name = "hello_world"
+   version = "0.1.0"
+   authors = ["Bradford Chamberlain <brad@chamberlain.com>"]
+
+   [dependencies]
+   Curl = "1.0.0"
+
+**Registry**
+
+Eventually, we will want a service similar to Cargo's crates.io_ to serve as our
+registry. Due to the amount of effort behind such a task, it has been decided
+that this can come at a later stage.
+
+The initial mason registry would be a GitHub repository containing a list of
+versioned manifest files from packages, not too unlike that of the OS X
+Homebrew package manager registry.
+
+The registry would follow a hierarchy as follows:
+
+::
+
+  https://github.com/chapel-lang/mason-registry
+
+  registry/
+    Curl/
+        1.0.0.toml
+        2.0.0.toml
+    RecordParser/
+        1.0.0.toml
+        1.1.0.toml
+        1.2.0.toml
+    VisualDebug/
+        2.2.0.toml
+        2.2.1.toml
+
+Each versioned manifest file would be identical to the manifest file in the
+top-level directory of the package repository, with one additional field that
+is not required in the repository manifest, a URL pointing to the repository
+and revision in which the version is located.
+
+Continuing the example from before, the 'registry' ``Mason.toml`` would include
+the additional ``source`` field:
+
+
+.. code-block:: ini
+
+   [package]
+   name = "hello_world"
+   version = "0.1.0"
+   authors = ["Brad Chamberlain <brad@chamberlain.com>"]
+   source = { git = "https://github.com/bradcray/hello_world", tag = "0.1.0" }
+
+   [dependencies]
+   Curl = "1.0.0"
+
+
+There are some recognized downsides of pursuing a GitHub repository as a registry:
+
+* It is not be feasible to cache packages centrally, compromising the
+  reliability of the entire package ecosystem.
+
+  * The packages themselves are distributed despite there being a central
+    registry of manifests.
+  * We can *not* guarantee that a given package version will always remain
```

Comment:
[Doesn't seem to matter](http://english.stackexchange.com/questions/204006/cannot-vs-can-not), but I fixed it.